src/main/scala/is/hail/expr/Type.scala

```diff
@@ -1184,4 +1232,24 @@ case class TStruct(fields: IndexedSeq[Field]) extends Type {
       }
     })
   }
+
+  lazy val byteOffsets: Array[Int] = {
+    val a = new Array[Int](size)
+    val missingBytes = UnsafeAnnotations.missingBytes(size)
+    var offset = missingBytes
+    fields.foreach { f =>
+      val fSize = f.typ.byteSize
+      val fAlignment = f.typ.alignment
+      val mod = offset % fAlignment
+      if (mod != 0)
+        offset += (fAlignment - mod)
+      a(f.index) = offset
+      offset += fSize
+    }
+    a
+  }
+
+  override lazy val byteSize: Int = if (size == 0) 0 else byteOffsets.last + fields.last.typ.byteSize
+
+  override def alignment: Int = 8
```

Comment:
Ah!  Yes.  The missing bits are another field.  Should we store missing bits as bytes?  Only can I can see where it matters is small structs of Booleans.  Probably doesn't matter.